sys/sleepq.c

```diff
@@ -227,10 +238,11 @@ sq_wakeup_t _sleepq_wait(void *wchan, const void *waitpt, sq_wakeup_t sleep) {
 /* Remove a thread from the sleep queue and resume it. */
 static bool sq_wakeup(thread_t *td, sleepq_chain_t *sc, sleepq_t *sq,
                       sq_wakeup_t wakeup) {
-  /* Only sq_enter sets TDF_SLPINTR flag and it holds the same sleepq spinlock
+  /* Only sq_enter sets TDF_SLP... flags and it holds the same sleepq spinlock
    * as sq_wakeup. Hence it's safe to read flags without holding thread's
    * spinlock. */
-  if (!(td->td_flags & TDF_SLPINTR) && (wakeup == SQ_ABORT)) {
+  if ((!(td->td_flags & TDF_SLPINTR) && (wakeup == SQ_ABORT)) ||
```

Comment:
>Could we split it into two?

Surely. I think we could also make long booleans readable by moving the `||` operator from the end of line to the beginning of the next one.

>Remember we can abort a timed sleep (signal delivery before timeout).

The `TDF_SLPINTR` flag is set when mode is at least `SQ_ABORT` so that should work.